### ISSUE TITLE
BMP image no longer works with latest release (Redhat)

### ISSUE THREAD

**Issue by noorez** [neutrinolabs/xrdp#2343]
It appears that the BMP image no longer works (others do) when I updated my installation of xrdp via the Redhat repos. Version 0.9.19. Is there any way to verify the validity of the BMP image?

I am not able to upload the BMP here to verify it appears

The error from the logs:
[20220812-09:02:05] [DEBUG] We try to load the following background file: /usr/share/xrdp/8-abstract-dark-2560x1440.bmp
[20220812-09:02:05] [ERROR] Error loading /usr/share/xrdp/8-abstract-dark-2560x1440.bmp [Unknown error]

**Comment by matt335672** [comment]
Hi @noorez 

I believe xrdp is built against imlib2 on that platform. You can confirm with `xrdp -v`.

The `unknown error` is actually returned from imlib2 with `IMLIB_LOAD_ERROR_UNKNOWN`.

A couple of thoughts:-
- You could use the `feh` utility from EPEL which also uses imlib2 to try to load the image. It may give you more information.
- Try loading the image in the gimp or similar and see if you get additional info.

HTH

**Comment by matt335672** [comment]
@noorez - do you need any more help with this?

**Comment by mmohylko** [comment]
For those who may find this helpful. Try using the **file** command to check the file type and compare it to the kind of the original file /usr/share/xrdp/xrdp_logo.bmp.
I found that the filetype should be 
```
$ file /usr/share/xrdp/xrdp_logo.bmp
/usr/share/xrdp/xrdp_logo.bmp: PC bitmap, Windows 3.x format
```
My custom bmp image created in Gimp was identified as the data type:
```
$ file /usr/local/share/xrdp/xrdp.bmp
/usr/local/share/xrdp/xrdp.bmp: data
```
That led to the same error. I had to export it with a "Compatibility option" =>"Do not write color space information" unset to make it Windows 3.x format.
